substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/NativeImageGenerator.java

```diff
@@ -848,7 +848,9 @@ private void setupNativeImage(String imageName, OptionValues options, Map<Method
                 prepareLibC();
 
                 CCompilerInvoker compilerInvoker = CCompilerInvoker.create(tempDirectory());
```

Comment:
How about:
```
boolean nativeToolchainNeed = !"llvm".equals(SubstrateOptions.CompilerBackend.getValue());
if (nativeToolchainNeed) {
    CCompilerInvoker compilerInvoker = CCompilerInvoker.create(tempDirectory());
    compilerInvoker.verifyCompiler();
    ImageSingletons.add(CCompilerInvoker.class, compilerInvoker);
}
```
Not sure it makes sense to bring `ExitAfterRelocatableImageWrite` in. Even with ExitAfterRelocatableImageWrite you still need a native toolchain for query code compilation, or am I missing something?

Instead of `"llvm".equals(SubstrateOptions.CompilerBackend.getValue())` it'd use `ImageSingletons.contains(LLVMFeature.class)`

Reply:
we use pre-generated CAP files and CAPcache for cross-platform targets, so there is no compiler needed.